app/coffee/HttpController.coffee

```diff
@@ -68,6 +69,20 @@ module.exports = HttpController =
 			res.send 400 # Bad Request
 			return
 
+		if lines.length > Settings.max_doc_lines
```

Comment:
When introducing a _new_ setting it's probably worth doing `Settings.max_doc_lines > 0  && lines.length > Settings.max_doc_lines` in case someone forgets to update a config file, so there isn't any unexpected behaviour when it is unset.  But I think we should defer the line length check to a separate PR for further consideration.

Reply:
Thanks. I was assuming that `settings.defaults.coffee` is still included in production builds for things that aren't overridden by the production settings. Is that not the case?

Reply:
It would (currently) need to be added in chef, and it is easy to forget to do that.  In future the settings file will always be used (except for web which still has its own custom settings file in kubernetes!)